allensdk/brain_observatory/ecephys/ecephys_project_cache.py

```diff
@@ -225,15 +225,35 @@ def get_units(self, suppress=None, **kwargs):
 
         """
 
+    def get_units(self, suppress=None, filter_by_validity: bool = True, **unit_filter_kwargs) -> pd.DataFrame:
+        """Reports a table consisting of all sorted units across the entire extracellular electrophysiology project.
+
+        Parameters
+        ----------
+        suppress : Optional[list], optional
+            A list of dataframe column names to hide, by default None
+            (None will hide dataframe columns specified in: SUPPRESS_FROM_UNITS)
+
+        filter_by_validity : bool, optional
+            Filter units so that only 'valid' units are returned, by default True
+
+        **unit_filter_kwargs :
+            Additional keyword arguments that can be used to filter units (for power users).
+
+        Returns
+        -------
+        pd.DataFrame
+            A table consisting of sorted units across the entire extracellular electrophysiology project
+        """
         if suppress is None:
             suppress = list(self.SUPPRESS_FROM_UNITS)
 
-        units = self._get_annotated_units(**kwargs)
+        units = self._get_annotated_units(filter_by_validity=filter_by_validity, **unit_filter_kwargs)
         units.drop(columns=suppress, inplace=True, errors="ignore")
 
         return units
 
-    def get_session_data(self, session_id):
+    def get_session_data(self, session_id, filter_by_validity: bool = True):
```

Comment:
did you want to add a `api_kwargs` parameter here so that the other options can be passed through?

Reply:
@njmei I think doing this would address @jsiegle's request